### PR TITLE
feat(msteams): Add personal notifications

### PR DESCRIPTION
**Pull request by 97varun**
Personal notifications for Microsoft Teams.
* Add msteams to list of available providers.
* Add choice of msteams in notification settings and provide defaults.
* Register `send_notification_as_msteams` as notification provider for msteams.

### CODE REVIEW

File: src/sentry/notifications/defaults.py [hunk 1]
```diff
@@ -36,5 +36,6 @@
 
 NOTIFICATION_SETTING_DEFAULTS = {
     ExternalProviders.EMAIL: NOTIFICATION_SETTINGS_ALL_SOMETIMES,
-    ExternalProviders.SLACK: NOTIFICATION_SETTINGS_SLACK,
+    ExternalProviders.SLACK: NOTIFICATION_SETTINGS_INSTANT_MESSAGING,
+    ExternalProviders.MSTEAMS: NOTIFICATION_SETTINGS_INSTANT_MESSAGING,
```
Comment (scefali) [inline src/sentry/notifications/defaults.py:40]: @97varun  So we decided to make MSTeams disabled by default?

Comment (97varun) [reply src/sentry/notifications/defaults.py:40]: To start with I think yes, I'm keeping it the same as how it would be for Slack.

Comment (scefali) [reply src/sentry/notifications/defaults.py:40]: @97varun So we made a change to make all new users have Slack by default. I need to backfill the old accounts before I can change the default, however.

**Review by NisanthanNanthakumar — APPROVED** [review]

File: tests/sentry/notifications/utils/test_get_setting_mapping_from_mapping.py [hunk 2]
```diff
@@ -113,6 +118,7 @@ def test_get_setting_mapping_from_mapping_slack_always(self):
         assert mapping == {
             ExternalProviders.EMAIL: NotificationSettingOptionValues.ALWAYS,
             ExternalProviders.SLACK: NotificationSettingOptionValues.ALWAYS,
+            ExternalProviders.MSTEAMS: NotificationSettingOptionValues.NEVER,
```
[the remark this answers is not in the document]
Comment (97varun) [reply tests/sentry/notifications/utils/test_get_setting_mapping_from_mapping.py:121]: Done!

**Review by NisanthanNanthakumar — APPROVED** [review]

**Pull request merged**